packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_liq_pool_add_liquidity.test.ts

```diff
@@ -0,0 +1,172 @@
+    }
+
+    const txn = await builder.liqPool.addLiquidity(addLiquidity, script)
+    const outs = await sendTransaction(container, txn)
+
+    expect(outs.length).toEqual(2)
+    const encoded: string = OP_CODES.OP_DEFI_TX_POOL_ADD_LIQUIDITY(addLiquidity).asBuffer().toString('hex')
+    // OP_RETURN + DfTx full buffer
+    const expectedRedeemScript = `6a${encoded}`
+    expect(outs[0].value).toEqual(0)
+    expect(outs[0].scriptPubKey.hex).toEqual(expectedRedeemScript)
+
+    // change
+    const change = await findOut(outs, providers.elliptic.ellipticPair)
+    expect(change.value).toBeLessThan(1)
+    expect(change.value).toBeGreaterThan(1 - 0.001) // deducted fee
+    expect(change.scriptPubKey.hex).toBe(`0014${HASH160(destPubKey).toString('hex')}`)
+    expect(change.scriptPubKey.addresses[0]).toBe(Bech32.fromPubKey(destPubKey, 'bcrt'))
+
+    // updated balance
+    const account = await jsonRpc.account.getAccount(await providers.getAddress())
+    expect(account.length).toStrictEqual(3)
+    expect(account).toContain('97.66000000@DFI')
+    expect(account).toContain('909.13000000@CAT')
+
+    // minted LM token
+    const lmTokenFound = account.find(tb => tb.split('@')[1] === 'DFI-CAT')
+    expect(lmTokenFound).toBeTruthy()
+  })
+
+  it('should reject invalid addLiquidity arg - more than 1 in `from`', async () => {
+    const script = await providers.elliptic.script()
+    await expect(builder.liqPool.addLiquidity({
+      from: [{
+        script,
+        balances: [{
+          token: 0,
+          amount: new BigNumber(10)
+        }]
+      }, {
+        script,
+        balances: [{
+          token: 1,
+          amount: new BigNumber(10)
+        }]
+      }],
+      shareAddress: script
+    }, script)).rejects.toThrow('`addLiquidity.from` array length must be ONE')
+  })
+
+  it('should reject invalid addLiquidity arg - not 2 TokenBalance in `from.balances`', async () => {
+    const script = await providers.elliptic.script()
+    await expect(builder.liqPool.addLiquidity({
+      from: [{
+        script,
+        balances: [{
+          token: 0,
+          amount: new BigNumber(10)
+        }, {
+          token: 0,
+          amount: new BigNumber(10)
+        }, {
+          token: 1,
+          amount: new BigNumber(10)
+        }]
+      }],
+      shareAddress: script
+    }, script)).rejects.toThrow('`addLiquidity.from[0].balances` array length must be TWO')
+  })
+
+  it('should reject invalid addLiquidity arg - `from.balances` is not a valid pair', async () => {
+    const script = await providers.elliptic.script()
+    await expect(builder.liqPool.addLiquidity({
+      from: [{
+        script,
+        balances: [{
+          token: 1,
+          amount: new BigNumber(10)
+        }, {
+          token: 1,
+          amount: new BigNumber(10)
+        }]
+      }],
+      shareAddress: script
+    }, script)).rejects.toThrow('`addLiquidity.from[0].balances` must consists of TWO different token')
+  })
+})
```

Comment:
Do we want to check our poolshares here too with `listpoolshares`

Reply:
Good point, adding them. Isn't 100% familiar with existing test tool/util, should be quick change since the tool existed.

Reply:
Added.